frontend/src/features/LLM/MwasHypothesisGenerator.tsx

```diff
@@ -0,0 +1,129 @@
+                pageStart: 0,
+                pageEnd: 1000,
+                identifiers: identifiers,
+            });
+        } catch (error) {
+            console.error(error);
+        }
+    };
+    console.log(hypothesisData);
+    const renderPlaceholder = () => {
+        if (isLoadingHypothesis) {
+            return <Skeleton variant='text' width={'100%'} height={60} />;
+        }
+        return null;
+    };
+    const hypothesisTextIsNonEmpty = () => hypothesisData && hypothesisData?.text?.length > 0;
+
+    const processHypothesisText = (text) => {
+        const combinedRegex = /PRJ\w+|\|([^|]+)\|/g;
+        const combinedMatches = text.match(combinedRegex);
+        if (combinedMatches) {
+            const finalComponents: (string | JSX.Element)[] = [];
+            let lastIndex = 0;
+            combinedMatches.forEach((match, index) => {
+                const matchIndex = text.indexOf(match, lastIndex);
+                if (matchIndex > lastIndex) {
+                    finalComponents.push(text.substring(lastIndex, matchIndex));
+                }
+                if (/PRJ\w+/.test(match)) {
+                    finalComponents.push(
+                        <Link
+                            key={`link-${index}`}
+                            href={`https://www.ncbi.nlm.nih.gov/bioproject/${match}`}
+                            target='_blank'
+                        >
+                            {match}
+                        </Link>,
+                    );
+                } else if (/\|([^|]+)\|/.test(match)) {
+                    finalComponents.push(
+                        <span key={`bold-${index}`} style={{ fontWeight: 'bold' }}>
+                            {match.slice(1, -1)}
+                        </span>,
+                    );
+                }
+                lastIndex = matchIndex + match.length;
+            });
+            if (lastIndex < text.length) {
+                finalComponents.push(text.substring(lastIndex));
+            }
+            return finalComponents;
+        }
+        return text;
+    };
+
+    return (
+        <Box>
+            <Box
+                sx={{
+                    zIndex: 10,
+                    flex: 1,
+                    display: 'flex',
+                    justifyContent: 'flex-end',
+                    height: 45,
+                    mr: hypothesisTextIsNonEmpty() ? 0 : '2%',
+                }}
+            >
+                <Tooltip title='Generate MWAS Hypotheses' placement='bottom'>
+                    <IconButton style={{ backgroundColor: 'rgba(86, 86, 86, 0.4)' }} onClick={onButtonClick}>
+                        <BlurOnIcon
+                            style={{
+                                color: '#9be3ef',
+                                fontSize: 30,
+                            }}
+                        />
+                    </IconButton>
+                </Tooltip>
```

Comment:
Can refactor to `LLMGeneratorButton.tsx` with appropriate props

Reply:
Created generic button component for LLM generation under `features/LLM/GenerateButton.tsx` see latest commit: https://github.com/serratus-bio/open-virome/commit/d56d4bbb08682c5672d2ba3d502e6597fc476ab6